Nucleation_model_sandbox.py

```diff
@@ -0,0 +1,129 @@
+#%%
+import datetime as dt
+import shutil
+import math as m
+import numpy as np
+import pandas as pd
+import random as ran
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+from scipy.integrate import solve_ivp
+#%%
+"""=========================== Constants ==========================="""
+
+k_B = 1.380649E-23 #J K-1 // Boltzmann's constant
+ep_0 = 8.854E-12  #F m-1 // electric permittivity of a vacuum
+N_a = 6.022E23 #mol-1 // Avogadro's number
+Rbar = 8.3145 #J mol-1 K-1 // Ideal constant
+
+"""======================== User system variables ========================="""
+#Parameters that can be changed during the trials
+T = 25 + 273.15 #K // Temperature
+C_LiO2_0 = 0.15 #mol m-3 // Electrolyte Concentration of LiO2 particles
+C_Li_0 = 0.15 #mol m-3 // Electrolyte Concentration Li+ particles
+Elyte_v = 1E-10 #uL3 // Electrolyte Volume
+N_0 = 0 # // Nucleations
+R_0 = 0
+A_0 = m.pi*0.009**2 #m2 // Cross sectional area
+time = 5000 #s // simulation time
+atol = 1e-8 # absolute tolerance
+rtol = 1e-4 # relative tolerance
+
+"""======================== User thermodyanic imputs ========================"""
+#parameters that depend on the system being studied
+
+C_LiO2_sat = 0.1 #mol m-3 // saturated concentration of LiO2 //Yin (2017)
+C_Li_sat = 0.1 #mol m-3 // saturated concentration of LiO2
+k_nu = 1E-6 #mol s-1 m-2 // kinetic rate constant for the nucleation reaction // Yin (2017)
+k_surf = 1E-8 #mol s-1 m-2 // kinetic rate constant for the growth reaction // Yin (2017)
+k_surf_des = 1E-8 #mol s-1 m-2 // kinetic rate constant for the growth reaction // Yin (2017)
+gamma_surf = 7.7E-3 #J m-2 //  surface energy of the newly formed crystal phase// Danner (2019) - should be replaced for LiS system
+sig_surf = 0.75 # J m-2 // Specific surface energy of Li2O2 with the electrolyte // Yin (2017)
+V = 1.98E-5 #m3 mol-1 // molar volume// Yin (2017)
+theta = 30*m.pi/180 # radians // Contact angle // Danner (2019) - should be replaced for LiS system
+D_LiO2 = 1.2E-9 #m2 -s-1 // diffusion ceofficient // Yin (2017)
+
+
+"""======================== Thermodyanic system cals ========================"""
+Elyte_v_SI = Elyte_v*1.0E-9 #
+phi = (2+m.cos(theta))*(1-m.cos(theta))**2/4  # - // contact angle correction factor
+
+"""========================== Initialization step =========================="""
+#%%
+#intializing the solution vector
+initial = {}
+initial['N'] = N_0
+initial['A'] = A_0
+initial['C_Li'] = C_Li_0
+initial['C_LiO2'] = C_LiO2_0
+sol_vec = list(initial.values())  # solution vector
+print(sol_vec)
+array = np.linspace(0, 1e-6, 500)
```

Comment:
A couple thoughts:
1. More descriptive names for `array` and `array2` would probably be helpful.
2. If I do understand them correctly, the smallest radius considered is 0 m, and the largest is 1 micron; is that correct?  If so, we probably want some `r_min` and likely a larger `r_max.`

Edit: so I see below that the critical radius is a feature of the electrolyte concentration... is this correct?  If so , then the current approach likely makes sense.  1 micrson as max radius still seems small, though.